site/app/models/gradeable/GradedComponent.php

```diff
@@ -51,35 +60,27 @@ class GradedComponent extends AbstractModel {
      * @param Core $core
      * @param Component $component The component this grade is associated with
      * @param User $grader The user who graded this component
-     * @param int[] $mark_ids The mark ids this graded component received
      * @param array $details any remaining properties
      * @throws \InvalidArgumentException if any of the details are invalid, or the component/grader are null
      */
-    public function __construct(Core $core, Component $component, User $grader, array $mark_ids, array $details) {
+    public function __construct(Core $core, Component $component, User $grader, array $details) {
```

Comment:
Would be nice to have TaGradedGradeable in here as well so you don't have to pass around both in the db methods.